File: C#/0896. Monotonic Array.cs
public class Solution {
    public bool IsMonotonic(int[] A) {
        if(A.Length==1){return true;}
        int left=1;
        while (left<A.Length && A[left]==A[left-1]){
            left+=1;
        }
        if (left==A.Length){return true;}
        
        else if(A[left]>A[left-1]){
            for(int i=left;i<A.Length;i++){
                if (A[i]<A[i-1]){
                    return false;
                }
            }
            return true;
        }
        else{
            for(int i=left;i<A.Length;i++){
                if (A[i]>A[i-1]){
                    return false;
                }
            }
            return true;
        }
    }
}

File: archives/20190519python/0896. Monotonic Array.py
class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        if len(A)<=1:
            return True
        if A[0]<A[1]:
            for i in range(1,len(A)):
                if A[i-1]>A[i]:
                    return False
        elif A[0]>A[1]:
            for i in range(1,len(A)):
                if A[i-1]<A[i]:
                    return False
        else:
            return self.isMonotonic(A[1:])
        return True
